var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -640,6 +649,10 @@ def edit(self, spec, prefix):
             if cuda_arch == "35" and spec.satisfies("+cuda_arch_35_k20x"):
                 gpuver = "K20X"
 
+            if spec.satisfies("@2023.2: +cusolvermp"):
```

Comment:
```suggestion
            if spec.satisfies("+cusolvermp"):
```
since the variant definition already includes the version constraint.